crates/air_r_formatter/src/r/auxiliary/call_arguments_table.rs

```diff
@@ -0,0 +1,508 @@
+use crate::r::auxiliary::call_arguments::FormatRCallArguments;
+use crate::{prelude::*, r::auxiliary::argument::fmt_argument_fields};
+use air_r_syntax::{
+    AnyRExpression, AnyRValue, RArgument, RArgumentList, RCallArguments, RDoubleValue,
+    RIntegerValue, RLanguage, RSyntaxNode, RSyntaxToken, RUnaryExpression,
+};
+
+use biome_formatter::{CstFormatContext, FormatElement, RemoveSoftLinesBuffer, format_args, write};
+use biome_rowan::{AstSeparatedElement, AstSeparatedList};
+
+const DOT_WIDTH: usize = 1;
+
+#[derive(Debug, Clone)]
+struct TableInfo {
+    rows: Vec<Vec<ArgData>>,
+    cols: Vec<ColumnInfo>,
+    remaining: Vec<AstSeparatedElement<RLanguage, RArgument>>,
+}
+
+#[derive(Debug, Clone)]
+struct ArgData {
+    node: RArgument,
+    kind: ArgKind,
+    separator: Option<RSyntaxToken>,
+    is_last_in_list: bool,
+}
+
+#[derive(Debug, Clone)]
+enum ArgKind {
+    Numeric {
+        integer_width: usize,
+        fractional_width: Option<usize>,
+    },
+    Other {
+        text: String,
+    },
+}
+
+#[derive(Debug, Default, Clone)]
+struct ColumnInfo {
+    // It's possible to have `has_decimal = true` and `max_fractional_part = 0`
+    // at the same time, if there is an argument like `1.`
+    has_decimal: bool,
+    max_width: usize,
+    max_value_width: usize,
+    max_integer_part: usize,
+    max_fractional_part: usize,
+}
+
+impl FormatRCallArguments {
+    pub(crate) fn fmt_table(
+        &self,
+        node: &RCallArguments,
+        f: &mut RFormatter,
+    ) -> FormatResult<Option<()>> {
+        let l_token = node.l_paren_token()?;
+        let r_token = node.r_paren_token()?;
+        let args = node.items();
+
+        if args.is_empty() {
+            return self.fmt_call_like(node, f).map(Some);
+        }
+
+        // Get table and alignment info
+        let table = match build_table(&args, f)? {
+            Some(table) => table,
+            None => return Ok(None),
+        };
+        let rows = table.rows;
+        let column_info = table.cols;
+
+        // Format with alignment
+        let formatted_table = format_with(|f| {
+            for (row_i, row) in rows.iter().enumerate() {
+                if row_i > 0 {
+                    // Rows are separated with hard line breaks because the
+                    // lines of arguments in table calls should never be
+                    // rearranged by the formatter.
+                    write!(f, [hard_line_break()])?;
+                }
+
+                for (col_j, arg_data) in row.iter().enumerate() {
+                    let (left_pad, right_pad) = if column_info[col_j].max_width > 0 {
+                        column_info[col_j].padding(&arg_data.kind)
+                    } else {
+                        // Empty columns are not padded, so that commas stick to
+                        // each other as in regular calls: `list(,,,)`
+                        (0, 0)
```

Comment:
Is this necessary anymore?

Reply:
yes but the comment is now slightly misleading, it's now:

```rs
                        // For empty columns don't add any incompressible whitespace
```